### ISSUE TITLE
onErrorResumeNext and onErrorReturn are missing error value

### ISSUE THREAD

**Issue by tomaszpolanski** [ReactiveX/rxdart#142]
Would it be possible to use `onErrorResumeNext` and `onErrorReturn` with not only the value of item to emit, like this:
``` Dart
new Observable.error(new Exception())
  .onErrorReturn(1)
```
but also with a function that passes the error:
``` Dart
new Observable.error(new Exception())
  .onErrorReturn((error) => error is Exception ? 1 : 0 )
```
This functionality is super userful in other versions of Rx.

**Comment by frankpepermans** [comment]
Certainly, we could do something in the line of the recent buffer/window update, where you pass a special handler.

I.e. buffer onStream/onTime/onFuture etc...
In this case onErrorReturn useValue/useHandler

... with the ability to create your own implementation of useX if needed

**Comment by tomaszpolanski** [comment]
I think having `onErrorResumeNext` that passes the error would be a nice have, but also it's not too hard to have a slightly different [implementation](https://gist.github.groupondev.com/tpolanski/3bd534176bb2e34f49e3fbaf81883aaf) of `OnErrorResumeNextStreamTransformer` - just putting it here if some else is in need of it.
Instead using it like this
``` Dart
  new Observable.error(new Exception())
      .transform(new OnErrorResumeNextStreamTransformer(new Observable.fromIterable([1, 2, 3])))
      .listen(print);
```
one could use it like this
``` Dart
  new Observable.error(new Exception())
      .transform(new OnErrorResumeNextStreamTransformer2((error) => new Observable.fromIterable([1, 2, 3])))
      .listen(print);
```

**Comment by henrik-d** [comment]
The link to the implementation is broken. Could you post it here again?

**Comment by tomaszpolanski** [comment]
@henrik-d Could you try out [this](https://gist.github.com/tomaszpolanski/824b19de4797c6edd6c4bf833b13c883) link?

**Comment by brianegan** [comment]
@tomaszpolanski Hey, sorry about the delay here. Found some time to add these to the RxDart library -- please checkout the #152 Pull Request and please feel free to leave feedback :)

**Comment by brianegan** [comment]
Published as part of 0.17.0
